api/src/main/java/org/openmrs/module/addresshierarchy/config/ConfigLoaderUtil.java

```diff
@@ -0,0 +1,405 @@
+	 *            "subdir/config.xml"
+	 * @return An empty string if the checksum hasn't changed, the new checksum otherwise.
+	 */
+	public static String getChecksumIfChanged(String configDirPath, String configFilePath) {
+		String savedChecksum = readLatestChecksum(configDirPath, configFilePath);
+		String checksum = computeChecksum(configDirPath, configFilePath);
+		return savedChecksum.equals(checksum) ? "" : checksum;
+	}
+	
+	/**
+	 * @see #getChecksumIfChanged(String, String)
+	 */
+	public String getChecksumIfChanged(String configFilePath) {
+		return getChecksumIfChanged(configDirPath, configFilePath);
+	}
+	
+	/**
+	 * Fetches all the files in a directory based on their extension.
+	 * 
+	 * @param configDirPath The absolute path to the config. directory, eg.
+	 *            "../configuration/addresshierarchy".
+	 * @param extension The extension to filter for, eg "xml".
+	 * @param recursive Set to true to continue recursively into subdirectories.
+	 * @return The list of {@link File} instances.
+	 */
+	public static List<File> getFiles(String configDirPath, String extension, boolean recursive) {
+		
+		final List<File> allFiles = new ArrayList<File>();
+		
+		final File[] files = new File(configDirPath).listFiles(getExtensionFilenameFilter(extension));
+		if (files != null) {
+			allFiles.addAll(Arrays.asList(files));
+		}
+		
+		if (recursive) {
+			final File[] dirs = new File(configDirPath).listFiles(getDirectoryFilenameFilter());
+			if (dirs != null) {
+				for (File dir : dirs) {
+					allFiles.addAll(getFiles(dir.getPath(), extension, true));
+				}
+			}
+		}
+		
+		return allFiles;
+	}
+	
+	/**
+	 * @see #getFiles(String, String, boolean)
+	 */
+	public static List<File> getFiles(String configDirPath, String extension) {
+		return getFiles(configDirPath, extension, false);
+	}
+	
+	/**
+	 * @see #getFiles(String, String, boolean)
+	 */
+	public List<File> getFiles(String extension, boolean recursive) {
+		return getFiles(configDirPath, extension, recursive);
+	}
+	
+	/**
+	 * @see #getFiles(String, String)
+	 */
+	public List<File> getFiles(String extension) {
+		return getFiles(configDirPath, extension);
+	}
+	
+	/**
+	 * Fetches the config. file from its relative path inside the configuration folder.
+	 * 
+	 * @param configDirPath The absolute path to the containing directory, eg.
+	 *            "../configuration/addresshierarchy"
+	 * @param configFilePath The config. file relative path inside the containing directory, eg.
+	 *            "subdir/config.xml"
+	 * @return The {@link File} instance.
+	 */
+	public static File getFile(String configDirPath, String configFilePath) {
+		StringBuilder path = new StringBuilder(configDirPath);
+		path.append(File.separator).append(configFilePath);
+		return new File(path.toString());
+	}
+	
+	/**
+	 * @see #getFile(String, String)
+	 */
+	public File getFile(String configFilePath) {
+		return getFile(configDirPath, configFilePath);
+	}
+	
```

Comment:
I'm not entirely sure what all of these filename filters and methods are for, but it might be simpler to save the latest checksum as a global property.  If there are multiple of these, then I can see why it would make sense to use files as we are, though if there are a fixed number of files to checksum, then we might also just have a fixed number of checksum files to manage by name.